src/main/java/bio/terra/dao/DatasetTableDao.java

```diff
@@ -1,17 +1,90 @@
 package bio.terra.dao;
 
+import bio.terra.configuration.DataRepoJdbcConfiguration;
+import bio.terra.metadata.Column;
+import bio.terra.metadata.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
 import org.springframework.stereotype.Repository;
 
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
 @Repository
 public class DatasetTableDao extends TableDaoBase {
 
+    private static final Logger logger = LoggerFactory.getLogger(DatasetTableDao.class);
+
+    private static final String sqlInsertTable = "INSERT INTO dataset_table " +
+        "(name, dataset_id, primary_key) VALUES (:name, :dataset_id, :primary_key)";
+
+    private final DataSource jdbcDataSource;
     private final NamedParameterJdbcTemplate jdbcTemplate;
 
+
     @Autowired
-    public DatasetTableDao(NamedParameterJdbcTemplate jdbcTemplate) {
-        super(jdbcTemplate, "dataset_table", "dataset_column", "dataset_id");
-        this.jdbcTemplate = jdbcTemplate;
+    public DatasetTableDao(DataRepoJdbcConfiguration dataRepoJdbcConfiguration) {
+        super(dataRepoJdbcConfiguration, "dataset_table", "dataset_column", "dataset_id");
+        this.jdbcDataSource = dataRepoJdbcConfiguration.getDataSource();
+        this.jdbcTemplate = new NamedParameterJdbcTemplate(this.jdbcDataSource);
```

Comment:
Why are you building a new JdbcTemplate here? We have one configured in ApplicationConfiguration that you can autowire. And that way, you can pass it in on `super()` and the base class doesn't have to make one.